js/ui/html_editor/ui/formDialog.js

```diff
@@ -115,6 +111,19 @@ class FormDialog {
         }, this._popupUserConfig);
     }
 
+    onHiding() {
+        this.beforeAddButtonAction = undefined;
+        this.deferred.reject();
+    }
```

Comment:
Minor comment.

Maybe, should we use the arrow function here to forget about the bind method?

```suggestion
    onHideHandler = () => {
        this.beforeAddButtonAction = undefined;
        this.deferred.reject();
    };
```

If we do it, we can remove the bind from the [63 line](https://github.com/DevExpress/DevExtreme/pull/23313/files#diff-4bbf331b9d53d10107a84b8e2775e58d7bf433a48e9cb2b6d44de4e684a550b6R63)

```ts
  this._popup.on('hiding', this.onHideHandler);
```